packages/app/src/commands/modernize.js

```diff
@@ -0,0 +1,162 @@
+ *   --log-missing-keys
+ */
+const path = require('path')
+const fs = require('fs')
+
+const { namespace } = require('@dhis2/cli-helpers-engine')
+const { reporter } = require('@dhis2/cli-helpers-engine')
+
+const { checkRequirements } = require('./modernize/checkRequirements.js')
+const { deleteLegacyFiles } = require('./modernize/deleteLegacyFiles.js')
+const {
+    generateTranslationMappings,
+} = require('./modernize/generateTranslationMappings.js')
+const {
+    createNewTranslationFiles,
+} = require('./modernize/createNewTranslationFiles.js')
+
+const CONSUMING_ROOT = path.join(process.cwd())
+const TRANSLATION_IN_DIR = path.join(CONSUMING_ROOT, 'src/i18n')
+const TRANSLATION_OUT_DIR = path.join(CONSUMING_ROOT, 'i18n')
+const CREATION_DATE = new Date().toISOString()
+
+const fileIsOldTranslationFile = fileName => fileName.match(/\.properties$/)
+
+const builder = {
+    primaryLanguage: {
+        describe: 'Primary language',
+        type: 'string',
+        default: 'en',
+    },
+
+    languages: {
+        describe:
+            'A comma separated list of languages to create files for, when omitted, all are used',
+        type: 'string',
+    },
+
+    deleteOld: {
+        describe:
+            'Deletes the old translation files, use `--no-delete-old` to keep the old files',
+        type: 'boolean',
+        default: 'true',
+    },
+
+    outDir: {
+        describe: 'Output directory for new translation files',
+        type: 'string',
+        default: TRANSLATION_OUT_DIR,
+    },
+
+    inDir: {
+        describe: 'Input directory where old translation files can be found',
+        type: 'string',
+        default: TRANSLATION_IN_DIR,
+    },
+
+    overrideExistingFiles: {
+        describe: 'Overriding the contents of existing translation files',
+        type: 'boolean',
+    },
+
+    appendToExistingFiles: {
+        describe:
+            'Appends the new contents to existing translation files, can only be used when not using `--override-existing-files`',
+        type: 'boolean',
+        conflicts: 'overrideExistingFiles',
+    },
+
+    logMissingKeys: {
+        describe:
+            'Log keys that are present in translation files which are not present in translation file of --primary-language',
+        type: 'boolean',
+        default: 'false',
+    },
+
+    deleteOldFiles: {
+        describe:
+            'Delete the old files that were transformed (will only delete files specified with the `--language` option when present)',
+        type: 'boolean',
+    },
+
+    pootlePath: {
+        describe: 'Set the path for the pootle server',
+        type: 'string',
+    },
+}
+
+const handler = ({
+    inDir,
+    outDir,
+    languages,
+    pootlePath,
+    deleteOldFiles,
+    logMissingKeys,
+    primaryLanguage,
+    overrideExistingFiles,
+    appendToExistingFiles,
+}) => {
+    const languagesToTransform = languages ? languages.split(/,\s*/) : []
+
+    const translationFiles = fs
```

Comment:
This throws a pretty nasty error if `inDir` doesn't exist.  We're checking for existence later in `checkRequirements` but those checks probably need to be moved here.

See https://github.com/dhis2/cli/pull/126/files#diff-f0dae67115941bb5d80b603159a41f59R111-R115